o3de/o3de
# Prefab - Add new unit tests for delete undo nodes

Signed-off-by: Junhao Wang <wjunhao@amazon.com>

## What does this PR do?

This PR adds 4 new unit tests for delete undo nodes:
1. PrefabUndoDeleteEntity (for source editing)
2. PrefabUndoInstanceLink (for source editing)
3. PrefabUndoDeleteAsOverride

Follow-up for: https://github.com/o3de/o3de/pull/13171

#### Changes:

PrefabUndoDeleteTests
1. PrefabUndoDeleteTests_DeleteEntity
1. PrefabUndoDeleteTests_DeleteNestedInstance
1. PrefabUndoDeleteTests_DeleteEntityAsOverride
1. PrefabUndoDeleteTests_DeleteInstanceAsOverride

PrefabTestFixture
1. Add helper functions for finding entity alias and nested instance alias in owning instance
1. Add helper functions for validating entity/instance existence
1. Move setting registry interface to PrefabTestFixture

_New helper functions should be generic and can be reused in other prefab unit tests, so I put them in PrefabTestFixture._

## How was this PR tested?

Passed all AzToolsFramework unit tests.
